client/nginx.go

```diff
@@ -890,7 +927,11 @@ func (client *NginxClient) getIDOfHTTPServer(upstream string, name string) (int,
 }
 
 func (client *NginxClient) get(path string, data interface{}) error {
-	ctx, cancel := context.WithTimeout(context.Background(), client.ctxTimeout)
+	return client.getWithContext(context.Background(), path, data)
+}
+
+func (client *NginxClient) getWithContext(ctx context.Context, path string, data interface{}) error {
+	ctx, cancel := context.WithTimeout(ctx, client.ctxTimeout)
```

Comment:
Not sure if creating a new context here is the best thing to do. It might be better to allow the user of the library to create the context themselves. That would allow them to have different timeout values for different requests if they wanted to. Think it would add more flexibility to the end user.

Reply:
Just thinking, if the user doesn't set the ctxTimeout, then it defaults to 0? I think this was a quick workaround for functions without a context being passed. Now that this PR updates that, then the condition you outlined is possible. I'm happy to remove the context.WithTimeout here and deprecate the client.ctxTimeout in this PR. Alternatively, I could add this to the non context based calls and leave the xWithContext functions use the context passed?

Reply:
@sjberman @lucacome thoughts?

Reply:
I agree with @dhurley. This function doesn't need to wrap the context with a new one, it can just use the context that's passed in, and the caller can decide what that context is.

Per your last comment, `get()` could use ctxTimeout and create the context to pass to this function.

Related: https://github.com/nginxinc/nginx-plus-go-client/issues/205

Reply:
thanks, was waiting for alignment. Just the pushed the update to this

Reply:
Do we need both `get` and `getWithContext` (as an example)? Is it just to keep backward compatibility? If so I would be okay with a breaking changes release

Reply:
It is just to keep backward compatibility. There are still some references calling the get, delete, patch and post that have not been refactored. I'm trying to limit changes in this PR, we could do those in a separate PR and bump the major version of the client?

Reply:
Sounds good